File: Benchmarking/Javascript/fib.ts

const fib_recursive = (n: number): number => {
  if(n <= 1) return n

  return fib_recursive(n-1) + fib_recursive(n-2)
}

let args = process.argv

if(args.length < 3) throw new Error("One parameter expected");

let n = parseInt(args[2])
if(n === NaN) throw new Error("Number format Incorrect")

let res = fib_recursive(n)
console.log(res)

File: Benchmarking/C++/fib.cpp
#include <iostream> 
using namespace std;

uint64_t fib_recursive(unsigned long long n){
    if(n<=1){
        return n;
    }
    return fib_recursive(n-1) + fib_recursive(n-2);
}

int main(int argc, char *argv[]){
    if(argc<2){
        throw std::invalid_argument("One parameter expected");
    }
    int n = atoi(argv[1]);
    uint64_t res = fib_recursive(n);
    cout<< res <<endl;
    return 0;
}